src/scenic/syntax/veneer.py

```diff
@@ -111,1000 +132,1341 @@
+    if not isinstance(scale, (builtins.int, builtins.float)):
+        raise TypeError('"mutate X by Y" with Y not a number')
+    for obj in objects:
+        if not isinstance(obj, Object):
+            raise TypeError('"mutate X" with X not an object')
+        obj.mutationScale = scale
 
 ### Prefix operators
 
 def Visible(region):
-	"""The :grammar:`visible <region>` operator."""
-	region = toType(region, Region, '"visible X" with X not a Region')
-	return region.intersect(ego().visibleRegion)
+    """The :grammar:`visible <region>` operator."""
+    region = toType(region, Region, '"visible X" with X not a Region')
+    return region.intersect(ego().visibleRegion)
 
 def NotVisible(region):
-	"""The :grammar:`not visible <region>` operator."""
-	region = toType(region, Region, '"not visible X" with X not a Region')
-	return region.difference(ego().visibleRegion)
+    """The :grammar:`not visible <region>` operator."""
+    region = toType(region, Region, '"not visible X" with X not a Region')
+    return region.difference(ego().visibleRegion)
 
 # front of <object>, etc.
 ops = (
-	'front', 'back', 'left', 'right',
-	'front left', 'front right',
-	'back left', 'back right'
+    'front', 'back', 'left', 'right',
+    'front left', 'front right',
+    'back left', 'back right', 'top',
+    'bottom', 'top front left', 'top front right',
+    'top back left', 'top back right',
+    'bottom front left', 'bottom front right',
+    'bottom back left', 'bottom back right'
 )
 template = '''\
 def {function}(X):
-	"""The :grammar:`{syntax} of <object>` operator."""
-	if not isinstance(X, Object):
-		raise RuntimeParseError('"{syntax} of X" with X not an Object')
-	return X.{property}
+    """The :grammar:`{syntax} of <object>` operator."""
+    if not isinstance(X, Object):
+        raise TypeError('"{syntax} of X" with X not an Object')
+    return X.{property}
 '''
 for op in ops:
-	func = ''.join(word.capitalize() for word in op.split(' '))
-	prop = func[0].lower() + func[1:]
-	definition = template.format(function=func, syntax=op, property=prop)
-	exec(definition)
+    func = ''.join(word.capitalize() for word in op.split(' '))
+    prop = func[0].lower() + func[1:]
+    definition = template.format(function=func, syntax=op, property=prop)
+    exec(definition)
 
 ### Infix operators
 
 def FieldAt(X, Y):
-	"""The :grammar:`<VectorField> at <vector>` operator."""
-	if not isinstance(X, VectorField):
-		raise RuntimeParseError('"X at Y" with X not a vector field')
-	Y = toVector(Y, '"X at Y" with Y not a vector')
-	return X[Y]
-
-def RelativeTo(X, Y):
-	"""The :scenic:`X relative to Y` polymorphic operator.
-
-	Allowed forms::
-
-		<value> relative to <value> # with at least one a field, the other a field or heading
-		<vector> relative to <oriented point> # and vice versa
-		<vector> relative to <vector>
-		<heading> relative to <heading>
-	"""
-	xf, yf = isA(X, VectorField), isA(Y, VectorField)
-	if xf or yf:
```

Comment:
Why not define `xf` and `yf` before the `if` as before and not have to redundantly call `isA`?